claim/claim.c

```diff
@@ -85,19 +84,26 @@ void claim_agent(char *claiming_arguments)
 
 void load_claiming_state(void)
 {
-    info("The claiming feature is under development and still subject to change before the next release");
-    return;
+    if (claimed_id != NULL) {
+        freez(claimed_id);
+        claimed_id = NULL;
+    }
 
     char filename[FILENAME_MAX + 1];
     struct stat statbuf;
 
-    snprintfz(filename, FILENAME_MAX, "%s/claim.d/is_claimed", netdata_configured_user_config_dir);
+    snprintfz(filename, FILENAME_MAX, "%s/claim.d/claimed_id", netdata_configured_user_config_dir);
     // check if the file exists
     if (lstat(filename, &statbuf) != 0) {
         info("File '%s' was not found. Setting state to AGENT_UNCLAIMED.", filename);
-        claiming_status = AGENT_UNCLAIMED;
    } else {
-        info("File '%s' was found. Setting state to AGENT_CLAIMED.", filename);
-        claiming_status = AGENT_CLAIMED;
+        FILE *f = fopen(filename,"rt");
+        claimed_id = callocz(1,statbuf.st_size+1);
+        size_t bytes_read = fread(claimed_id,1,statbuf.st_size,f);
+        claimed_id[bytes_read] = 0;
+        info("File '%s' was found. Setting state to AGENT_CLAIMED using agent_id=.", filename, claimed_id);
```

Comment:
fclose?

Reply:
If fread returns 0 (empty file) should we still be unclaimed (claimed_id == NULL) to prevent publishing to topics `agent//meta` etc. etc.?

Reply:
Will push an improvement in a bit

Reply:
No error message if fopen returns NULL (valid case for an unclaimed agent). The other points are valid.

Reply:
Hi, but this is case where lstat was successful and then open fails. We can change the message to info